modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APIEndpointRelevantObjectEntryModelListener.java

```diff
@@ -174,16 +174,17 @@ private void _validate(ObjectEntry objectEntry) {
 							"x-can-have-a-maximum-of-255-alphanumeric-" +
 								"characters";
 					}
-					else if (!StringUtil.isLowerCase(pathString)) {
-						message = "%s must contain only lower case characters";
-						messageKey =
-							"x-must-contain-only-lower-case-characters";
-					}
 					else {
 						message = "%s must start with the \"/\" character";
 						messageKey = "x-must-start-with-the-x-character";
 					}
 
+					if (!StringUtil.isLowerCase(pathString)) {
```

Comment:
I think it does not make sense to have it in a different if clause, as we will override the value of `message` & `messageKey`. Maybe we could update the conditions to ensure the order does not matter:

```
if (!pathString.startsWith(StringPool.FORWARD_SLASH)) {
	message = "%s must start with the \"/\" character";
	messageKey = "x-must-start-with-the-x-character";
}
else if (!StringUtil.isLowerCase(pathString)) {
	message = "%s must contain only lower case characters";
	messageKey =
		"x-must-contain-only-lower-case-characters";
}
else {
	message =
		"%s can have a maximum of 255 alphanumeric " +
			"characters";
	messageKey =
		"x-can-have-a-maximum-of-255-alphanumeric-" +
			"characters";
}
```

Reply:
First sorry for not put you as a reviewer because I forgot that :man_facepalming: . However, I made in a separate way because Brian committed a sort commit in which he put like this:

```java
if (pathString.startsWith(StringPool.FORWARD_SLASH)) {
      message =
	      "%s can have a maximum of 255 alphanumeric " +
		      "characters";
      messageKey =
	      "x-can-have-a-maximum-of-255-alphanumeric-" +
		      "characters";
      }
      else if (!StringUtil.isLowerCase(pathString)) {
      message = "%s must contain only lower case characters";
      messageKey =
	      "x-must-contain-only-lower-case-characters";
      }
      else {
      message = "%s must start with the \"/\" character";
      messageKey = "x-must-start-with-the-x-character";
}
```
Sorting the content of the message and the messageKey

Reply:
This is the commit: https://github.com/brianchandotcom/liferay-portal/commit/8ab237bf92d8fe5b606c65179a2135332892cfd5

Reply:
Please check the code that I included, as the conditions are different, so the messages in each if are different

Reply:
The conditions in the original code are not properly being checked, as the first if clause is checking a valid condition, the path must start wih slash, so it does not make sense to check if it starts with slash. What it make sense is to check if it does NOT start with slash. And, if you don't find the cause, then by default you suppose it must be greater than 255 characters